### DISCUSSION TITLE
Consider setting project_urls in setup.py to allow tools to find the source and changelog

### DISCUSSION THREAD

**Discussion by strayer** [Chia-Network/chia-blockchain#11509] [Ideas]
Tools like Renovate utilize the `project_urls` key in `setup.py` to find the project source and changelog when checking pypi releases, for example:

https://github.com/renovatebot/renovate/blob/098d22d09b099667bd96c2594a79098d0fdbe01a/lib/modules/datasource/pypi/index.ts#L114-L144

```typescript
    if (dep.info?.project_urls) {
      for (const [name, projectUrl] of Object.entries(dep.info.project_urls)) {
        const lower = name.toLowerCase();

        if (
          !dependency.sourceUrl &&
          (lower.startsWith('repo') ||
            lower === 'code' ||
            lower === 'source' ||
            githubRepoPattern.exec(projectUrl))
        ) {
          dependency.sourceUrl = projectUrl;
        }

        if (
          !dependency.changelogUrl &&
          ([
            'changelog',
            'change log',
            'changes',
            'release notes',
            'news',
            "what's new",
          ].includes(lower) ||
            changelogFilenameRegex.exec(lower))
        ) {
          // from https://github.com/pypa/warehouse/blob/418c7511dc367fb410c71be139545d0134ccb0df/warehouse/templates/packaging/detail.html#L24
          dependency.changelogUrl = projectUrl;
        }
      }
    }
```

Setting project_urls as `{"Source": "https://github.com/Chia-Network/chia-blockchain/", "Changelog": "https://raw.githubusercontent.com/Chia-Network/chia-blockchain/main/CHANGELOG.md"}` should be enough to help Renovate find the release notes. (just keep in mind that I didn't actually test this and only followed the documentation)

This is great for people using Renovate to keep their repositories up-to-date since one can quickly check what changed directly in the dependency bump PRs instead of having to hunt for changelogs manually.

https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#project-urls

**Comment by altendky** [comment]
Thanks for pointing this out.  I'm certainly on board with having any of the metadata we're missing filled in.  Would you be interested in providing a PR for that?  If not I'll try to take a look at some point.  It's been awhile since I've set this stuff up, but https://github.com/altendky/qtrio/blob/43c7ff24c0be2f7a3df86eef3c6cc5dae2f7ffd3/setup.cfg#L8-L16 is one example I've got.  It is in `setup.cfg`, so the syntax is different, but the form ought to be pretty similar.

**Reply by strayer** [reply]
Sure, I’ll take a shot at it. Will take a bit though because I have to figure out a way to test this. Probably publishing on the pypi test environment and telling Renovate to use that instead.

**Reply by altendky** [reply]
Cool, thanks.  Let me know if you have any questions.

**Reply by strayer** [reply]
FYI: I started work on this, but am facing problems. Hopefully the Renovate people can help me figure it out. (https://github.com/renovatebot/renovate/discussions/15769 for reference)

**Reply by altendky** [reply]
Thanks for the update and link.  I subscribed so I can keep an eye on their thoughts.
